quantus/metrics/faithfulness/faithfulness_correlation.py

```diff
@@ -387,7 +328,59 @@ def evaluate_batch(
         scores_batch:
             The evaluation results.
         """
-        return [
-            self.evaluate_instance(model=model, x=x, y=y, a=a)
-            for x, y, a in zip(x_batch, y_batch, a_batch)
-        ]
+        # Prepare shapes. Expand a_batch if not the same shape
+        if x_batch.shape != a_batch.shape:
+            a_batch = np.broadcast_to(a_batch, x_batch.shape)
+
+        # Flatten the attributions.
+        batch_size = a_batch.shape[0]
+        a_batch = a_batch.reshape(batch_size, -1)
+        n_features = a_batch.shape[-1]
+
+        # Predict on input.
+        x_input = model.shape_input(
+            x_batch, x_batch.shape, channel_first=True, batched=True
+        )
+        y_pred = model.predict(x_input)[np.arange(batch_size), y_batch]
+
+        pred_deltas = []
+        att_sums = []
+
+        x_batch_shape = x_batch.shape
+        # For each test data point, execute a couple of runs.
+        for i_ix in range(self.nr_runs):
+            # Randomly mask by subset size.
+            a_ix = np.stack(
+                [
+                    np.random.choice(n_features, self.subset_size, replace=False)
+                    for _ in range(batch_size)
+                ],
+                axis=0,
+            )
+            x_perturbed = self.perturb_func(
+                arr=x_batch.reshape(batch_size, -1),
+                indices=a_ix,
+            )
+            x_perturbed = x_perturbed.reshape(*x_batch_shape)
+
+            # Check if the perturbation caused change
+            for x_element, x_perturbed_element in zip(x_batch, x_perturbed):
+                warn.warn_perturbation_caused_no_change(
+                    x=x_element, x_perturbed=x_perturbed_element
+                )
+
+            # Predict on perturbed input x.
+            x_input = model.shape_input(
+                x_perturbed, x_batch.shape, channel_first=True, batched=True
+            )
+            y_pred_perturb = model.predict(x_input)[np.arange(batch_size), y_batch]
+            pred_deltas.append(y_pred - y_pred_perturb)
+
+            # Sum attributions of the random subset.
+            att_sums.append(a_batch[np.arange(batch_size)[:, None], a_ix].sum(axis=-1))
+        pred_deltas = np.stack(pred_deltas, axis=1)
+        att_sums = np.stack(att_sums, axis=1)
+
+        similarity = self.similarity_func(a=att_sums, b=pred_deltas, batched=True)
```

Comment:
Isn't `batch_baseline_replacement_by_indices` always batched? Why do we need `batched=True` argument?

@annahedstroem have you ever used a different `similarity_func` here?

Reply:
Here, the `batched=True` argument goes into a similarity function (for example `correlation_pearson`), not the `batch_baseline_replacement_by_indices`. Similarity functions can be batched and not batched (at the moment at least) so this argument is needed here.